api/controllers/v1/github.ts

```diff
@@ -109,3 +114,101 @@ export const TriggerBuild = async (event: any = {}, context: any = {}): Promise<
     body: 'Job Queued',
   };
 };
+
+/**
+ * Deletes build artifacts for a given project + branch combination.
+ * @param event
+ */
+export const MarkBuildArtifactsForDeletion = async (event: APIGatewayEvent) => {
+  const consoleLogger = new ConsoleLogger();
+  const defaultHeaders = { 'Content-Type': 'text/plain' };
+
+  const ghEventType = event.headers['X-GitHub-Event'];
+  if (ghEventType !== 'pull_request') {
+    const errMsg = 'GitHub event type is not of type "pull_request"';
+    return {
+      statusCode: 400,
+      headers: defaultHeaders,
+      body: errMsg,
+    };
+  }
+
+  if (!validateJsonWebhook(event, c.get<string>('githubDeletionSecret'))) {
+    const errMsg = "X-Hub-Signature incorrect. Github webhook token doesn't match";
+    return {
+      statusCode: 401,
+      headers: defaultHeaders,
+      body: errMsg,
+    };
+  }
+
+  if (!event.body) {
+    const err = 'MarkBuildArtifactsForDeletion does not have a body in event payload';
+    consoleLogger.error('MarkBuildArtifactsForDeletion', err);
+    return {
+      statusCode: 400,
+      headers: defaultHeaders,
+      body: err,
+    };
+  }
+
+  let payload: PullRequestEvent | undefined;
+  try {
+    payload = JSON.parse(event.body) as PullRequestEvent;
```

Comment:
NIT: You're already doing runtime validation below to check and see if the action is `closed`, but I think it could be helpful to check and see if the request is actually for a PR. I believe you can have the `action` equal `closed` for an issue as an example. [Looking at the documentation](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request), it appears that you can check to see if the `pull_request` property exists on the payload.

Reply:
There's currently some logic above to check if the event's header is for the "pull_request" event.

```js
const ghEventType = event.headers['X-GitHub-Event'];
if (ghEventType !== 'pull_request') {
  const errMsg = 'GitHub event type is not of type "pull_request"';
  return {
    statusCode: 400,
    headers: defaultHeaders,
    body: errMsg,
  };
}
```

Just curious if we want to have checks for both, or if we should prioritize checking for the `pull_request` object over the header

Reply:
Oh, totally missed that!